config_test.go

```diff
@@ -0,0 +1,168 @@
+package timescaledb
+
+import (
+	"testing"
+	"time"
+
+	"go.k6.io/k6/lib/types"
+	"gopkg.in/guregu/null.v3"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_getConsolidatedConfig_succeeds(t *testing.T) {
```

Comment:
I would've done all tests as one big [table driven test](https://github.com/golang/go/wiki/TableDrivenTests). As in something like in the [influxdb output](https://github.com/grafana/k6/blob/cb6295a30ae3563919ae145c8b5e7799e342c653/output/influxdb/config_test.go)

Reply:
Right, I thought this might raise some eyebrows. I do know what table tests are and consciously made a decision not to use them in this situation. 
That said, the fact that I anticipated I would receive some criticism about it means that the reasons I have are not widely adopted. However, I do think that they’re valid.

I’m happy to discuss (debate 🙂  ) that further. There are a few of the tests which could be refactored into table tests if you feel strongly about it, but for now I am of the opinion that individual tests are always acceptable whereas table tests are sometimes acceptable, and not the other way around.

**tldr: I think that table tests are less readable.**

Here are some reasons why I didn't use a table test:
- each test is small enough for logic to be conceptualized at once
- multiple functionality assertions in one test were avoided
- the tests do not requires the reader to jump around reading ancillary code
- My goal is to make it obvious why a test fails (often explicitly demonstrating what I do not care about in the test), not DRY.

This reasoning is from the experience I’ve had reading, writing, and maintaining tests and using unit tests to debug problems.

Reply:
@shirleyleu, in general, I agree with your reasons and this is a good summary:

> individual tests are always acceptable whereas table tests are sometimes acceptable, and not the other way around.

However, you're missing the other side of the arguments, why table tests are sometimes better:
- modification of the tests (e.g. after the tested code is refactored) is much easier when it's in a single place
- when the test code for a whole bunch of very similar tests (with different inputs and expected outputs) is in a single place, you're much more sure that there aren't any mistakes in the tests themselves; adding a new test case is very unlikely to contain a copy-paste error or omission
- sometimes, reading and understanding table tests is actually easier - you might not care about the test minutia all that much, but seeing a "table" of inputs and expected outputs can help someone understand both the test _and the tested code_ at a glance
- [sub-tests](https://pkg.go.dev/testing#hdr-Subtests_and_Sub_benchmarks) allow you to separate and differentiate every case in a table test, so it's easier to debug any failures

In this case, it doesn't really matter and I'm fine with the current tests, they're ok as separate tests. However, the function names are a bit unidiomatic because of the underscores. In general, [you should probably use `TestNameWithMixedCaps`](https://golang.org/doc/effective_go#mixed-caps). There are exceptions for related [example tests](https://pkg.go.dev/testing#hdr-Examples), or sometimes to separate out [related sub-tests](https://ieftimov.com/post/testing-in-go-naming-conventions/#function-naming-conventions), but then the underscores just separate out the `MixedCaps` parts (e.g. `TestGetConsolidatedConfig_Succeeds`, `TestGetConsolidatedConfig_FromEnvVariables`), not every word.

Reply:
Thanks for your feedback. 
Respectfully, I’m not entirely convinced by all the points, but I’m happy to conform to the team’s/maintainer's style if it matters. (I’m [not against table tests](https://github.com/grafana/k6/pull/2104/files#diff-83ace201a5a29b48a281ff8c41fadb87037a9cd4469ce2543ad234b86cf9aa3e), though I didn’t follow through with this commit.) 

If someone is just going to see this test file and vomit or have the irresistible urge to refactor, I can just do it now. Sorry @MStoykov that I didn’t just take the hint earlier.

OK! I will change the test case names!

Reply:
> If someone is just going to see this test file and vomit or have the irresistible urge to refactor, I can just do it now.

:smile: No, as I mentioned, the tests are fine as separate ones, don't worry about it! And while @MStoykov would have probably done them differently, I don't think he'd have an irresistible urge to refactor them later :stuck_out_tongue_closed_eyes: I'm reading his comment as minor nitpick, at least

Reply:
Great, thanks! 😌

I've replaced the mostly snake case with mostly mixed case now. I've preserved the first underscore to match the case of the function names which they are testing. 
Let me know if you want it fully mixed case no matter the case of the function name.